mne/connectivity/cfc.py

```diff
@@ -0,0 +1,529 @@
+            logger.info(n_surrogates - sur + 1)
+        # circular shift
+        sur_triggers = np.mod(trigger_inds + shifts[sur], n_samples)
+        sur_triggers = sur_triggers.astype(np.int)
+        sur_distributions[:, sur] = np.mean(ampmat[:, sur_triggers], axis=1)
+
+    sur_fits_mean = np.mean(sur_distributions, axis=1)
+    sur_fits_std = np.std(sur_distributions, axis=1)
+
+    ztraces = (traces - sur_fits_mean[:, None]) / sur_fits_std[:, None]
+
+    # Compute FDR threshold
+    p_values = stats.norm.pdf(ztraces)
+    accept, _ = fdr_correction(p_values, alpha=alpha)
+    # check if any acceptable values are found
+    if not accept.any():
+        raise RuntimeError('No significant cross frequency coupling found'
+                           ' try changing the frequency range.')
+    z_threshold = np.abs(ztraces[accept]).min()
+
+    return ztraces, z_threshold
+
+
+def phase_amplitude_coupling(data, sfreq, l_phase_freq, h_phase_freq,
+                             l_amp_freq, h_amp_freq, bin_num=18, method='fft',
+                             surrogates=False, n_jobs=1):
+    """ Compute modulation index for the given data.
+
+    Parameters
+    ----------
+    data : ndarray (n_epochs, n_times)
+        Signal time series.
+    sfreq : float
+        Sampling frequency in Hz.
+    l_phase_freq : float
+        Lower edge of phase modulating frequency.
+    h_phase_freq : float
+        Upper edge of phase modulating frequency.
+    l_amp_freq : float
+        Lower edge of amplitude modulated frequency.
+    h_amp_freq : float
+        Upper edge of amplitude modulated frequency.
+    bin_num : int
+        Number of phase bins. Default is 18. (20 degrees * 18 = 360)
+    method : str
+        Filter method used. Default is 'fft'.
+    n_jobs : int
+        Number of jobs used for parallelization. Default 1.
+    surrogates : bool
+        Return surrogates values for data specific to PAC,
+        the phase series and amplitude envelope are shuffled for every trial.
+        Default value is False.
+
+    Returns
+    -------
+    normalized_amplitude : ndarray, shape (n_epochs, bin_num)
+        The normalized amplitude values across each bin.
+    phase_bins : array, shape (number of bins + 1,)
+        Binned phase points. Used for plotting.
+
+    References
+    ----------
+    Measuring phase-amplitude coupling between neuronal oscillations
+    of different frequencies.
+    Tort ABL, Komorowski R, Eichenbaum H, Kopell N., (J Neurophysiol. 2010)
+    """
+
+    # obtained number of trials
+    n_epochs = len(data)
+    # time points in each epochs (assuming epochs of equal length)
+    n_times = data.size / n_epochs
+    # reshape epochs into single time series for filtering
+    filter_data = data.reshape((1, n_times * n_epochs))[0]
+
+    # Filter data into phase modulating freq and amplitude modulated freq.
+    filter_x_fp = band_pass_filter(filter_data, sfreq, l_phase_freq,
```

Comment:
Split the part where the filtering is done and put it into a different function?